### PR TITLE
Project setup (once again)

### PR DESCRIPTION
**Pull request by isomethane**

### CODE REVIEW

File: .github/workflows/gradle-build.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+          settings-path: ${{ github.workspace }} # location for the settings.xml file
+
+      - name: Publish
+        run: ./gradlew publishToMavenLocal --console=plain
+        
+      - name: Run tests
+        run: |
+          cd tests
+          ./gradlew test --console=plain
+          cd ..
```
Comment (nbirillo) [inline .github/workflows/gradle-build.yml]: Maybe` ./tests/gradlew test --console=plain` is better?

**Review by nbirillo — CHANGES_REQUESTED** [review]
Great!

File: .github/workflows/gradle-build.yml (hunk 1 above)
Comment (isomethane) [reply .github/workflows/gradle-build.yml]: This did not work for me, but I think adding working-directory is also fine.

**Review by nbirillo — APPROVED** [review]
Great!

**Pull request merged**